packages/core/strapi/lib/factories.d.ts

```diff
@@ -43,6 +43,6 @@ interface Router {
   routes: Route[];
 }
 
-export function createCoreRouter(uid: string, cfg: RouterConfig): () => Router;
-export function createCoreController(uid: string, cfg: ControllerConfig): () => Controller;
-export function createCoreService(uid: string, cfg: ServiceConfig): () => Service;
+export function createCoreRouter(uid: string, cfg?: RouterConfig = {}): () => Router;
+export function createCoreController(uid: string, cfg?: ControllerConfig = {}): () => Controller;
+export function createCoreService(uid: string, cfg?: ServiceConfig = {}): () => Service;
```

Comment:
for the 3 lignes, the typing is not good here
I'd suggest
```suggestion
export function createCoreRouter(uid: string, cfg?: ({strapi}) => RouterConfig): () => Router;
export function createCoreController(uid: string, cfg?:({strapi}) => ControllerConfig): () => Controller;
export function createCoreService(uid: string, cfg?: ({strapi}) => ServiceConfig): () => Service;
```
And for each we should have
`ControllerConfig = Partial<Controller>`

I can contribute if you want